src/ICeWall/src/ICeWall.tcl

```diff
@@ -3561,26 +3904,1336 @@ namespace eval ICeWall {
+      }
+      if {$scaled_rdl_width > $max_width} {
+        utl::error PAD 213 "Width set for RDL layer $rdl_layer_name ([ord::dbu_to_microns $scaled_rdl_width]) is greater than the maximum width of the layer in this technology ([ord::dbu_to_microns $max_width])"
+      }
+    } else {
+      dict set library rdl width [$rdl_layer getWidth]
+    }
+    set rdl_width [dict get $library rdl width]
+
+    if {[dict exists $library rdl spacing]} {
+      set scaled_rdl_spacing [ord::microns_to_dbu [dict get $library rdl spacing]]
+      set spacing [$rdl_layer getSpacing]
+
+      if {$scaled_rdl_spacing < $spacing} {
+        utl::error PAD 214 "Spacing set for RDL layer $rdl_layer_name ([ord::dbu_to_microns $scaled_rdl_spacing]) is less than the required spacing for the layer in this technology ([ord::dbu_to_microns $spacing])"
+      }
+    }  
+
+    if {[dict exists $library num_pads_per_tile]} {
+      check_num_pads_per_tile_option [dict get $library num_pads_per_tile]
+    } else {
+      utl::error PAD 215 "The number of pads within a bump pitch has not been specified"
+    }
+  }
+
+  proc add_pad {args} {
+    variable footprint
+    variable block
+    variable type_index
+    variable db
+    variable tech
+
+    set db [::ord::get_db]
+    set tech [$db getTech]
+    set block [ord::get_db_block]
+
+    set padcell_name {}
+
+    if {[llength $args] % 2 == 1} {
+        utl::error PAD 109 "Incorrect number of arguments for add_pad - expected an even number, received [llength $args]"
+    }
+
+    set padcell {}
+    # Determine the padcell name
+    if {[dict exists $args -name]} {
+      set padcell_name [dict get $args -name]
+    } else {
+      if {[dict exists $args -type]} {
+        set type [dict get $args -type]
+        if {![dict exists $type_index $type]} {
+          dict set type_index $type 0
+        } 
+        set idx [dict get $type_index $type]
+        set padcell_name "${type}_$idx"
+        incr idx
+        dict set type_index $type $idx
+      } else {
+        utl::error PAD 110 "Must specify -type option if -name is not spedified"
+      }
+    }
+    if {[dict exists $footprint padcell $padcell_name]} {
+      utl::error PAD 216 "A padcell with the name $padcell_name already exists"
+    }
+    dict set padcell name $padcell_name
+
+    set process_args $args
+    while {[llength $process_args] > 0} {
+      set arg [lindex $process_args 0]
+      set value [lindex $process_args 1]
+
+      switch $arg {
+        -name      {dict set padcell name $value}
+        -signal    {dict set padcell signal_name $value}
+        -edge      {dict set padcell side [check_edge_name $value]}
+        -type      {dict set padcell type [check_cell_type $value]}
+        -cell      {dict set padcell cell_name [check_cell_name $value]}
+        -location  {dict set padcell cell [check_location $value]}
+        -bump      {dict set padcell bump [check_bump $value]}
```

Comment:
This should report an error if the same bump is used more than once.

Reply:
Added consistency check to ensure that the same row/col combination is only used once